.github/PULL_REQUEST_TEMPLATE/release.md

```diff
@@ -0,0 +1,77 @@
+# Release vFIXME
+
+## Checklist
+
+- [ ] [Create a new github milestone](https://github.com/DataDog/dd-trace-go/milestones/new) for the next version and
+  move any open PRs and issues that are in the current release milestone to the new one.
+
+- [ ] Draft [the release note](https://github.com/DataDog/dd-trace-go/releases/new) and ask the teams for its review on
+  the `#guild-dd-go` channel.
+
+- Deploy a release candidate:
+    - Create a release candidate version on this release branch:
+        - [ ] Bump and commit the release candidate version string in `/internal/version/version.go` (eg. `v1.2.3-rc.4`)
```

Comment:
I'm assuming this involves a PR to be created and merged before doing the next step. Is that correct? (If so, can you state that, and provide a link to an example PR that does this)

Reply:
No PR merging at this point, all of this happens on the temporary release branch so this whole process is just a commit + tag + push on this branch.

Reply:
No PR merging at this point, all of this happens on the temporary release branch so this whole process is just a commit + tag + push on this branch.

Reply:
Now automated by https://github.com/DataDog/dd-trace-go/pull/1284/commits/c7fcf9fd43ff54cd030376b442d0249b8a639224